src/app/(sidebar)/transaction/build/components/Operations.tsx

```diff
@@ -0,0 +1,659 @@
+
+      const opErrors: OpBuildingError = {};
+
+      if (!op.operationType || hasErrors || hasMissingFields) {
+        const opLabel = TRANSACTION_OPERATIONS[op.operationType]?.label;
+        opErrors.label = `Operation #${idx + 1}${opLabel ? `: ${opLabel}` : ""}`;
+        opErrors.errorList = [];
+
+        if (!op.operationType) {
+          opErrors.errorList.push("Select operation type");
+        }
+
+        if (hasMissingFields) {
+          opErrors.errorList.push("Fill out all required fields");
+        }
+
+        if (hasErrors) {
+          const errorCount = Object.keys(op.error).length;
+
+          opErrors.errorList.push(
+            errorCount === 1 ? "Fix error" : "Fix errors",
+          );
+        }
+      }
+
+      if (!isEmptyObject(opErrors)) {
+        allErrorMessages.push(opErrors);
+      }
+    });
+
+    // Callback to the parent component
+    updateBuildIsValid({ operations: allErrorMessages.length === 0 });
+
+    return allErrorMessages;
+  };
+
+  const formErrors = getOperationsError();
+  const sourceAccountComponent = formComponentTemplate("source_account");
+
+  const OperationTabbedButtons = ({
+    index,
+    isUpDisabled,
+    isDownDisabled,
+    isDeleteDisabled,
+  }: {
+    index: number;
+    isUpDisabled: boolean;
+    isDownDisabled: boolean;
+    isDeleteDisabled: boolean;
+  }) => {
+    return (
+      <TabbedButtons
+        size="md"
+        buttons={[
+          {
+            id: "moveUp",
+            hoverTitle: "Move up",
+            icon: <Icon.ArrowUp />,
+            onClick: () =>
+              updateOptionParamAndError({
+                type: "move-before",
+                index,
+              }),
+            isDisabled: isUpDisabled,
+          },
+          {
+            id: "moveDown",
+            hoverTitle: "Move down",
+            icon: <Icon.ArrowDown />,
+            onClick: () =>
+              updateOptionParamAndError({
+                type: "move-after",
+                index,
+              }),
+            isDisabled: isDownDisabled,
+          },
+          {
+            id: "duplicate",
+            hoverTitle: "Duplicate",
+            icon: <Icon.Copy07 />,
+            onClick: () =>
+              updateOptionParamAndError({
+                type: "duplicate",
+                index,
+              }),
+          },
+          {
+            id: "delete",
+            hoverTitle: "Delete",
+            icon: <Icon.Trash01 />,
+            isError: true,
+            isDisabled: isDeleteDisabled,
+            onClick: () =>
+              updateOptionParamAndError({
+                type: "delete",
+                index,
+              }),
+          },
+        ]}
+      />
+    );
+  };
+
+  const OperationTypeSelector = ({
+    index,
+    operationType,
+  }: {
+    index: number;
+    operationType: string;
+  }) => (
+    <Select
+      fieldSize="md"
+      id={`${index}-operationType`}
+      label="Operation type"
```

Comment:
i think it's missing a description. The attached screenshot from figma includes a description, but I don't see one on the site. 
<img width="605" alt="Screenshot 2024-05-06 at 3 18 05 PM" src="https://github.com/stellar/laboratory/assets/3912060/b57d4a16-0a46-40e8-9ec5-e65df40ae0b1">

Reply:
Thanks for catching that! I'll add it in the next PR, as this one is already quite large.